src/QsFmt/App/Program.fs

```diff
@@ -56,29 +71,37 @@ let run arguments inputs =
                     match command source with
                     | Ok result ->
                         if input = "-" then printf "%s" result else File.WriteAllText(input, result)
-                        0
+                        { RunResult.Default with FilesProcessed = 1 }
                     | Error errors ->
                         // Change the "-" input to say "<Standard Input>" in the error
                         let input = if input = "-" then "<Standard Input>" else input
                         errors |> List.iter (eprintfn "%s, %O" input)
-                        1
+                        { RunResult.Default with ExitCode = 1; SyntaxErrors = errors }
             with
             | :? IOException as ex ->
                 eprintfn "%s" ex.Message
-                3
+                { RunResult.Default with ExitCode = 3 }
             | :? UnauthorizedAccessException as ex ->
                 eprintfn "%s" ex.Message
-                4
+                { RunResult.Default with ExitCode = 4 }
+
+    and foldResults (previousRunResult: RunResult) filePath =
+        let newRunResult = (filePath |> doOne arguments)
+        {
+            FilesProcessed = previousRunResult.FilesProcessed + newRunResult.FilesProcessed
+            ExitCode = max previousRunResult.ExitCode newRunResult.ExitCode
+            SyntaxErrors = previousRunResult.SyntaxErrors @ newRunResult.SyntaxErrors
+        }
 
     and doMany arguments inputs =
```

Comment:
Looks like the `arguments` parameter is not used here anymore. Actually, since `arguments` isn't modified anywhere under `run`, I think you could remove the `arguments` parameters from all the nested functions. All the references to `arguments` can just refer directly to the `run` function's `arguments` parameter.

Reply:
Suggestion implemented.